baseapp/abci.go

```diff
@@ -905,6 +906,9 @@ func (app *BaseApp) FinalizeBlock(req *abci.FinalizeBlockRequest) (res *abci.Fin
 		for _, streamingListener := range app.streamingManager.ABCIListeners {
 			if err := streamingListener.ListenFinalizeBlock(app.finalizeBlockState.Context(), *req, *res); err != nil {
 				app.logger.Error("ListenFinalizeBlock listening hook failed", "height", req.Height, "err", err)
+				if app.streamingManager.StopNodeOnErr {
+					os.Exit(1)
```

Comment:
We should exist better imho, this will skip of the Close() method on the app and could leave the db with opened connections

Reply:
I'll add app.Close() before 👌

Reply:
Sounds good, I'm returning errors now. On Commit it will trigger a `CONSENSUS FAILURE!!!`, but the node won't crash 🤔

Reply:
Oh yeah, forgot this issue with comet: https://github.com/cometbft/cometbft/issues/4054 :/